core/src/main/java/com/epam/esm/repository/impl/OrderRepositoryImpl.java

```diff
@@ -0,0 +1,126 @@
+import com.epam.esm.util.DateTimeUtil;
+import lombok.RequiredArgsConstructor;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+@RequiredArgsConstructor
+public class OrderRepositoryImpl implements OrderRepository {
+    private final JdbcTemplate jdbcTemplate;
+    private static final String CREATE_ORDER = "insert into \"order\" (cost, created_date, user_id) values (?, ?, ?)";
+    private static final String CREATE_ORDER_ITEM = "insert into order_item " +
+            "(quantity, order_id, gift_certificate_id) values (?, ?, ?)";
+    private static final String READ_ORDER_BY_ID =
+            "select \"user\".id, username, password, email, cost, created_date from \"order\" " +
+                    "join \"user\" on \"user\".id = \"order\".user_id and \"order\".id = ?";
+    private static final String READ_ORDER_BY_USER_ID =
+            "select id, cost, created_date from \"order\" where user_id = ?";
+    private static final String READ_ORDER_ITEM_BY_ORDER_ID = "select order_item.id, " +
+            "order_item.gift_certificate_id as certificateId, name, description, price, duration, quantity " +
+            "from order_item join gift_certificate on order_item.gift_certificate_id = gift_certificate.id " +
+            "where order_id =?";
+    private static final String UPDATE_COST = "update \"order\" set cost = ? where id = ?";
+    private static final String DELETE_ORDER = "delete from \"order\" where id = ?";
+    private static final String DELETE_ORDER_ITEMS = "delete from order_item where order_id = ?";
+    private final RowMapper<OrderItem> orderItemMapper;
+    private final RowMapper<User> userMapper;
+
+    @Override
+    public Order createOrder(Order order) {
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        LocalDateTime cratedDate;
+        cratedDate = DateTimeUtil.toZone(order.getCreatedDate(), TimeZoneConfig.DATABASE_ZONE, ZoneId.systemDefault());
+        jdbcTemplate.update(connection -> {
+            PreparedStatement preparedStatement =
+                    connection.prepareStatement(CREATE_ORDER, Statement.RETURN_GENERATED_KEYS);
+            int index = 1;
+            preparedStatement.setBigDecimal(index++, order.getCost());
+            preparedStatement.setTimestamp(index++, Timestamp.valueOf(cratedDate));
+            preparedStatement.setLong(index, order.getUser().getId());
+            return preparedStatement;
+        }, keyHolder);
+        Long orderId = ((Number) keyHolder.getKeys().get("id")).longValue();
```

Comment:
Look at ((Number) keyHolder.getKeys().get("id")).longValue(); Can you optimaze it?  

Why some times we get Map other times just value? 
keyHolder.getKeys() vc keyHolder.getKey()

        tag.setId((Long) keyHolder.getKeys().get("id"));
        tag.setId(keyHolder.getKey());